packages/legend-query/src/stores/QueryBuilderLambdaBuilder.ts

```diff
@@ -209,6 +221,66 @@ export const processPostFilterOnLambda = (
   return lambda;
 };
 
+export const buildPropertyExpressionChain = (
+  propertyExpression: ValueSpecification,
+  queryBuilderState: QueryBuilderState,
+): ValueSpecification | undefined => {
+  if (propertyExpression instanceof AbstractPropertyExpression) {
+    const newPropertyExpression =
+      propertyExpression.accept_ValueSpecificationVisitor(
+        new QueryBuilderValueSpecificionCloneHelper(
```

Comment:
As discussed, let's use a cheap copy strategy here. Since you we know this is just the property expression chain. My take is we can just traverse the chain and copy anything but property expression over. Property expressions are the only thing you need to `new` here. But I might be wrong as the situation can get complicated with `subtyping`, I'm not 100% sure about that.